src/gl.rs

```diff
@@ -274,6 +283,35 @@ pub fn tex_image_2d(target: GLenum,
     }
 }
 
+pub fn compressed_tex_image_2d(target: GLenum,
+                               level: GLint,
+                               internal_format: GLenum,
+                               width: GLsizei,
+                               height: GLsizei,
+                               border: GLint,
+                               image_size: GLsizei,
+                               data: &[u8]) {
+    unsafe {
+        ffi::CompressedTexImage2D(target, level, internal_format, width, height, border,
+                                  image_size, data.as_ptr() as *const GLvoid);
+    }
+}
+
+pub fn compressed_tex_sub_image_2d(target: GLenum,
+                                   level: GLint,
+                                   xoffset: GLint,
+                                   yoffset: GLint,
+                                   width: GLsizei,
+                                   height: GLsizei,
+                                   format: GLenum,
+                                   image_size: GLsizei,
+                                   data: &[u8]) {
```

Comment:
I'd probably remove the `image_size` parameter and use `data.len() as GLsizei` instead, since that parameter should be the size of the `data` buffer in bytes ([spec](https://www.opengl.org/sdk/docs/man/html/glCompressedTexSubImage2D.xhtml)), and it's arguably safer.